.github/workflows/build-lint-test.yml

```diff
@@ -33,7 +33,32 @@ jobs:
           node-version: ${{ matrix.node-version }}
           cache: 'yarn'
       - run: yarn --immutable --immutable-cache
-      - run: yarn build
+      - run: yarn build:source
+      - name: Require clean working directory
+        shell: bash
+        run: |
+          if ! git diff --exit-code; then
+            echo "Working tree dirty at end of job"
+            exit 1
+          fi
+
+  build-types:
+    name: Build types
+    runs-on: ubuntu-latest
+    needs:
+      - prepare
+    strategy:
+      matrix:
+        node-version: [16.x, 18.x, 20.x]
```

Comment:
We probably don't need to run build on all these Node versions. But can be fixed in another PR.

Reply:
Agreed, but out of scope for this PR.